tests/issues.rs

```diff
@@ -171,6 +171,40 @@ mod issue514 {
     }
 }
 
+#[cfg(feature = "serialize")]
+mod issue533 {
+    use serde_derive::Deserialize;
+    use std::io::BufReader;
+
+    #[derive(Deserialize)]
+    struct X {}
+
+    #[test]
+    fn test() {
+        let xml = "<!DOCTYPE X [<!-- -->]><X></X>";
+        let reader = BufReader::with_capacity(20, xml.as_bytes());
+        let _: X = quick_xml::de::from_reader(reader).unwrap();
+    }
+}
```

Comment:
Please put this test into `serde-issues.rs`. It compiles only when `serialize` feature is active so does not require feature gate.